### PR TITLE
Added main script that allows for Calendar integration, and misc. changes

### LINKED ISSUE #29: Google Calendar Automation
Automatically add events to 'PESU Competitive Coding' Google Calendar by polling various websites. We already know what the URL is going to be for the upcoming contest and can periodically look it up until it gives a 404.

Reasons for not taking up this issue:
1. It doesn't take long to add google calendar events manually, about 10 minutes a week.
2. Automating this could cause issues. Changes in schedule will anyway have to be addressed manually.

### PR DESCRIPTION
**Pull request by LaRuim**
#29 
@souravtecken 

This is a major commit, as a few changes overall were needed to facilitate the making of the main script.

1. Created a main script called **run.py**. This fetches events from the Google Calendar, calls the appropriate scraper and does all the File I/O operations. It also has functions to update the database. 

2. Altered db_tools.py slightly to allow each function to be called from run.py

3. Altered processor.py slightly to perform its task via the _process_ method, to allow it to be called from run.py

4. Removed unnecessary commented code and added explanatory comments over all files.

5. Slightly altered the way _logging_ is used over the entire project, standardizing it.

6. Edited the README to reflect the changes in the way the project is structured.

7. Edited requirements.txt to reflect dependencies better

**To-do:**
1. Make HackerRank and Google scrapers compatible with the main script.
2. Create a method to update README in the database folder after scraping.
3. More error handling
4. ?

Fixes #29

### CODE REVIEW

File: database/db_tools.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import csv
-import logging
 from os import listdir
 from os.path import join
 from collections import Counter
 from functools import lru_cache
 from typing import List, Set, Tuple, Dict, Callable, Any
 from tinydb import TinyDB, where
 from ratings import elo
+from utils.log import *
```
Comment (souravtecken) [inline database/db_tools.py]: I wouldn't recomending *, you could maybe do a `import utils.log as log` and use `log.info()`, so on. Could this be changed in the scrapers too?

File: database/db_tools.py [hunk 2]
```diff
@@ -144,18 +144,19 @@ def log_unmapped_handles(site_username_tuple_list: List[Tuple[str, str]]) -> Non
...
     """
     Removes unmapped handles from outdated rank files
     to reduce space and time it takes for the script to run
     """
-    with open(join(CONTEST_RANKS_DIR, file_name), 'r') as rank_file:
+    with open(file_name, 'r') as rank_file:
         input_data: str = rank_file.read()
-
-    with open(join(CONTEST_RANKS_DIR, file_name), 'w') as rank_file:
+    #print(len(input_data))
```
Comment (souravtecken) [inline database/db_tools.py]: Remove this comment, and what's the purpose of `count`?

File: scrapers/README.md [hunk 3]
```diff
@@ -4,12 +4,4 @@ This module contains scripts responsible for scraping contest pages.
...
-1. map_handle_to_usn.py: After a contest is scraped this one maps the handles to its respective USN so that even if the handle changes in the future, it does not affect the rank.
-1. codejam.py: Scraper for scoreboard for Google Codejam for its new interface that was launched in 2018. 
-2. codechef.py: Scraper for Codechef contests. This one uses regular expression to fetch the score board from the raw web page.
-3. hackearth.py: Scraper for Hackerearth contests. This scrapes the entire global leaderboard without any filter.
-
-#### Usage
-
-1. Modify the scoreboard url in google.py, Run it, make necessary changes in the scoreboard(Change default view to 20, so on...), Press enter to continue.
-2. Copy the raw scoreboard from codechef to a file say `codechef-contest-year.in` and run `python3 codechef.py codechef-contest-year.in` 
+**UNDER PROGRESS**
```
Comment (souravtecken) [inline scrapers/README.md:7]: Why, this is a branch anyway, you needn't touch the README, keep the content in it, modify it when you need to

**Review by souravtecken — CHANGES_REQUESTED** [review]

File: database/db_tools.py (hunk 1 above)
Comment (LaRuim) [reply database/db_tools.py]: Yup, done

File: database/db_tools.py (hunk 2 above)
Comment (LaRuim) [reply database/db_tools.py]: nothing, twas for debugging; has been removed

File: scrapers/README.md (hunk 3 above)
Comment (LaRuim) [reply scrapers/README.md:7]: uh, my bad; also really don't feel like putting the old contents back so I left it unchanged

Comment (souravtecken) [reply scrapers/README.md:7]: What is that supposed to mean xD

Comment (LaRuim) [reply scrapers/README.md:7]: I mean, I'll add the content back in a later commit xD

**Pull request merged**